File: offer_collector/shared_models/DynamicFilter.cs
using Offer_collector.Models;
using UnifiedOfferSchema;

namespace shared_models
{
    public class DynamicFilter
    {
        public List<Languages> Languages { get; set; } = new List<Languages>();
        public List<string> ExperienceLevels { get; set; } = new List<string>();
        public List<string> EducationNames { get; set; } = new List<string>();
    }
}

File: frontend/types/details/dynamicFilter.ts
import { language } from "../list/Offer/language";

export interface ApiDynamicFilters {
  languages: language[];
  experienceLevels: string[];
  educationNames: string[];
}

export type FilterValue = string | number;

export interface FilterItem {
  label: string;
  value: FilterValue;
}

export type FilterKey =
  | "languages"
  | "languageLevels"
  | "experienceMonths"
  | "experienceLevels";

export interface dynamicFilter {
  key: FilterKey;
  header: string;
  items: FilterItem[];
}